cnxdb/cli/subcommands.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+"""cnx-db subcommands"""
+from __future__ import print_function
+import sys
+from functools import reduce
+
+from .common import assign_db_args
+from .discovery import register_subcommand
+
+
+def _compile_connection_string_parts(args_namespace):
+    """Given an ``argparse.Namespace``, translate this to a connection string
+    parts (dict).
+
+    """
+    return {
+        'dbname': args_namespace.dbname,
+        'user': args_namespace.user,
+        # 'password': None,
+        'host': args_namespace.host,
+        'port': args_namespace.port,
+    }
+
+
+def _translate_parts_to_string(connection_string_parts):
+    """Translate the connection string parts ot a string"""
+    return reduce(
```

Comment:
Is it the same as ' '.join(['='.join(x) for x in connection_string_parts.items() if x[1] is not None])?